hummingbot/strategy/simple_trade/simple_trade.pyx

```diff
@@ -0,0 +1,299 @@
+                                                               price = quantized_price)
+                    self.logger().info("Limit buy order has been placed")
+
+                else:
+                    order_id = self.c_sell_with_specific_market(market_info,
+                                                                amount = quantized_amount,
+                                                                order_type = OrderType.LIMIT,
+                                                                price = quantized_price)
+                    self.logger().info("Limit sell order has been placed")
+
+                self._time_to_cancel[order_id] = self._current_timestamp + self._cancel_order_wait_time
+        else:
+            self.logger().info(f"Not enough balance to run the strategy. Please check balances and try again.")
+
+
+    cdef c_has_enough_balance(self, object market_info):
+        cdef:
+            MarketBase market = market_info.market
+            double base_asset_balance = market.c_get_balance(market_info.base_asset)
+            double quote_asset_balance = market.c_get_balance(market_info.quote_asset)
+            OrderBook order_book = market_info.order_book
+            double price = order_book.c_get_price_for_volume(True, self._order_amount).result_price
+
+        return quote_asset_balance >= self._order_amount * price if self._is_buy else base_asset_balance >= self._order_amount
+
+
+    cdef c_process_market(self, object market_info):
+        cdef:
+            MarketBase maker_market = market_info.market
+            set cancel_order_ids = set()
+
+        if self._place_orders:
+
+            #If current timestamp is greater than the start timestamp + time delay place orders
+            if self._current_timestamp > self._start_timestamp + self._time_delay:
+
+                self._place_orders = False
+                self.logger().info(f"Current time: "
+                                   f"{datetime.fromtimestamp(self._current_timestamp).strftime('%Y-%m-%d %H:%M:%S')} "
+                                   f"is now greater than "
+                                   f"Starting time: "
+                                   f"{datetime.fromtimestamp(self._start_timestamp).strftime('%Y-%m-%d %H:%M:%S')} "
+                                   f" with time delay: {self._time_delay}. Trying to place orders now. ")
+                self.c_place_orders(market_info)
+
+        active_orders = self.market_info_to_active_orders.get(market_info, [])
+
+        if len(active_orders) >0 :
+            for active_order in active_orders:
+                if self._current_timestamp >= self._time_to_cancel[active_order.client_order_id]:
+                    cancel_order_ids.add(active_order.client_order_id)
+
+        if len(cancel_order_ids) > 0:
+
+            for order in cancel_order_ids:
+                self.c_cancel_order(market_info, order)
+
+
```

Comment:
You should add event handlers and log messages to indicate when orders are filled.